modules/navier_stokes/src/fvbcs/INSFVTurbulentViscosityWallFunction.C

```diff
@@ -101,29 +104,35 @@ INSFVTurbulentViscosityWallFunction::boundaryValue(const FaceInfo & fi) const
     const ADReal b_c = 1 / NS::von_karman_constant *
                        (std::log(NS::E_turb_constant * std::max(wall_dist, 1.0) / mu) + 1.0);
     const ADReal c_c = parallel_speed;
+
     u_tau = (-b_c + std::sqrt(std::pow(b_c, 2) + 4.0 * a_c * c_c)) / (2.0 * a_c);
     y_plus = wall_dist * u_tau * rho / mu;
-    mut_log = rho * Utility::pow<2>(u_tau) * wall_dist / parallel_speed;
+    mu_wall = rho * Utility::pow<2>(u_tau) * wall_dist / parallel_speed;
+    mut_log = mu_wall - mu;
   }
   else if (_wall_treatment == "neq")
   {
     // Assign non-equilibrium wall function value
-    y_plus = std::pow(_C_mu, 0.25) * wall_dist * std::sqrt(_k(current_argument, state)) * rho / mu;
-    mut_log = mu * (NS::von_karman_constant * wall_dist /
-                        std::log(std::max(NS::E_turb_constant * y_plus, 1 + 1e-4)) -
-                    1.0);
+    y_plus =
+        std::pow(_C_mu, 0.25) * wall_dist * std::sqrt(_k(current_argument, old_state)) * rho / mu;
+    mu_wall = mu * (NS::von_karman_constant * y_plus /
+                    std::log(std::max(NS::E_turb_constant * y_plus, 1 + 1e-4)));
+    mut_log = mu_wall - mu;
   }
-
   if (y_plus <= 5.0)
     // sub-laminar layer
-    return mut_visc;
+    return 0.0;
   else if (y_plus >= 30.0)
     // log-layer
-    return mut_log + mut_visc;
+    return std::max(mut_log, 1e-12);
```

Comment:
The 1e-12 appears often as a limiter. Considering making a coded parameter in NS Namespace